aiohttp/web_request.py

```diff
@@ -510,19 +510,26 @@ def text(self):
         return bytes_body.decode(encoding)
 
     @asyncio.coroutine
-    def json(self, *, loads=json.loads, wrong_format=None, if_check=True):
-        """
-        Return BODY as JSON.
-        If the `Content-Type` is not `application/json`,
+    def json(self, *, loads=json.loads, allowed_types=None, err_handler=None):
+        """ Return BODY as JSON. If the `Content-Type` is not in allowed_types,
         then the request is rejected.
-        You can also handle it manually by passing the
-        `wrong_format` parameter.
+        You can also handle it manually by passing the `err_handler` parameter.
         """
-        if if_check and self.content_type != 'application/json':
-            if wrong_format is None:
+        wrong_format_flag = False
+        if isinstance(allowed_types, str):
```

Comment:
1. So, if backward compatibility is a very important in this question (I don't agree with that) than maybe force checking by default, but define some compatibility switch which turns old behavior. I mean some swith of form `aiohttp.DEFAULT_JSON_CT=None`. Like `logging.raiseExceptions=False`.
2. I'm `-1` for having API to pass list/tuple/container, since I can't imagine real case.